src/main/java/org/opentripplanner/middleware/triptracker/ManageTripTracking.java

```diff
@@ -0,0 +1,92 @@
+package org.opentripplanner.middleware.triptracker;
+
+import org.opentripplanner.middleware.models.TrackedJourney;
+import org.opentripplanner.middleware.persistence.Persistence;
+
+import java.util.UUID;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsInt;
+
+public class ManageTripTracking {
+
+    private ManageTripTracking() {
+    }
+
+    public static final int TRIP_TRACKING_UPDATE_FREQUENCY_SECONDS
+        = getConfigPropertyAsInt("TRIP_TRACKING_UPDATE_FREQUENCY_SECONDS", 5);
+
+    /**
+     * Start tracking by providing a unique journey id and tracking update frequency to the caller.
+     */
+    public static StartTrackingResponse startTrackingTrip(TrackingPayload payload) {
+
+        // Create unique journey id for this trip. If the same trip is repeated it will not clash with previous
+        // journeys of the same trip.
+        var journeyId = UUID.randomUUID().toString();
+
+        // Start tracking journey.
+        var trackedJourney = new TrackedJourney(journeyId, payload);
+        Persistence.trackedJourneys.create(trackedJourney);
+
+        // Provide response.
+        return new StartTrackingResponse(
+            TRIP_TRACKING_UPDATE_FREQUENCY_SECONDS,
+            getInstructions(TripStage.START),
+            journeyId,
+            getTripStatus(TripStage.START)
+        );
+    }
+
+    /**
+     * Update the tracking location information provided by the caller.
+     */
+    public static UpdateTrackingResponse updateTracking(TrackingPayload payLoad, TrackedJourney trackedJourney) {
+
+        // Update tracked journey.
+        trackedJourney.update(payLoad);
+        Persistence.trackedJourneys.replace(trackedJourney.id, trackedJourney);
```

Comment:
It is just the `locations` field that is being explicitly updated, correct? Try using `Persistence.update` instead and pass the list with all the locations, and this will also update the `lastUpdated` field (right now that field is not getting updated).